portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -30,21 +36,38 @@
   List<String> commentsList = new ArrayList<String>();
 
   @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Gets comment input from <form> by id selector
-    String comment = request.getParameter("comment-input");
-    commentsList.add(comment);
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      PreparedQuery results = datastore.prepare(query);
 
-    response.sendRedirect("/index.html");
-  }
+      for (Entity entity : results.asIterable()) {
+          String commentInQuery = (String) entity.getProperty("comment");
+
+          commentsList.add(commentInQuery);
+      }
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
       response.setContentType("application/json");
       String json = convertToJsonUsingGson(commentsList);
       response.getWriter().println(json);
   }
 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      // Gets comment input from <form> by id selector
+      String comment = request.getParameter("comment-input");
+      long timestamp = System.currentTimeMillis();
+      
+      Entity commentEntity = new Entity("Comment");
+      commentEntity.setProperty("comment", comment);
+      commentEntity.setProperty("timestamp", timestamp);
```

Comment:
Pull these three strings into a constant string in your class and use it here and also in `doGet` method. This way you don't run into typo problems.

```java
public class DataServlet extends HttpServlet {
  private static final String COMMENT_TABLE_NAME = "Comment";
  private static final String COMMENT_COLUMN_NAME = "comment";
  private static final String TIMESTAMP_COLUMN_NAME = "timestamp";
```
Also note that a column name `timestamp` is not very useful. It's better if the column name describes what is the value represents. For example in your case this is `submit_time`.